packages/embed-components/src/Theme/state/sagas.spec.ts

```diff
@@ -0,0 +1,280 @@
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+import ReduxSagaTester from 'redux-saga-tester'
+import {
+  createFactory,
+  destroyFactory,
+  registerThemeService,
+  getThemeService,
+} from '@looker/embed-services'
+import { Looker40SDK as LookerSDK } from '@looker/sdk'
+import type { ITheme } from '@looker/sdk'
+import type { IAPIMethods } from '@looker/sdk-rtl'
+import { session } from '../../test-utils'
+import {
+  defaultThemesState,
+  themeActions,
+  THEMES_SLICE_NAME,
+  themesSlice,
+} from './slice'
+import * as sagas from './sagas'
+
+describe('SelectTheme sagas', () => {
+  let sagaTester: ReduxSagaTester<any>
+  const sdk: IAPIMethods = new LookerSDK(session)
+
+  beforeEach(() => {
+    createFactory(sdk)
+    sagaTester = new ReduxSagaTester({
+      initialState: { [THEMES_SLICE_NAME]: defaultThemesState },
+      reducers: {
+        [THEMES_SLICE_NAME]: themesSlice.reducer,
+      },
+    })
+    sagaTester.start(sagas.saga)
+  })
+
+  describe('initSaga', () => {
+    const { initAction, initSuccessAction, setFailureAction } = themeActions
+
+    it('registers theme service and sends initSuccessAction on success', async () => {
+      expect(getThemeService).toThrow('Service ThemeService not found')
+
+      sagaTester.dispatch(initAction())
+
+      await sagaTester.waitFor('themes/initSuccessAction')
+      expect(getThemeService).not.toThrow()
+      const calledActions = sagaTester.getCalledActions()
+      expect(calledActions).toHaveLength(2)
+      expect(calledActions[0]).toEqual(initAction())
+      expect(calledActions[1]).toEqual(initSuccessAction())
+    })
+
+    it('sends setFailureAction on error', async () => {
+      // destroy the factory to cause initAction to fail
+      destroyFactory()
+
+      sagaTester.dispatch(initAction())
+
+      await sagaTester.waitFor('themes/setFailureAction')
+      const calledActions = sagaTester.getCalledActions()
+      expect(calledActions).toHaveLength(2)
+      expect(calledActions[0]).toEqual(initAction())
+      expect(calledActions[1]).toEqual(
+        setFailureAction({ error: 'Factory must be created with an SDK.' })
+      )
+    })
+  })
+
+  describe('loadThemeDataSaga', () => {
+    const {
+      loadThemeDataAction,
+      loadThemeDataSuccessAction,
+      setFailureAction,
+    } = themeActions
+    const lookerTheme = { id: 1, name: 'Looker' } as unknown as ITheme
+    const defaultTheme = { id: 2, name: 'custom_theme' } as unknown as ITheme
+    const anotherTheme = { id: 3, name: 'custom_theme_1' } as unknown as ITheme
```

Comment:
are the ids returned by the API actually numeric?

Reply:
Could the `as unknown` be required because the IDs are numeric? What if you use `as Partial<ITheme>`?